paddle/fluid/framework/details/broadcast_op_handle.cc

```diff
@@ -126,6 +126,9 @@ void BroadcastOpHandle::BroadcastOneVar(
             &VariableVisitor::GetMutableTensor(out_var));
       }
     });
+    for (auto &p : places_) {
```

Comment:
Why wait?

Reply:
In Reduce mode, all the updating of parameters are placed in separate GPUs, so we should use broadcast the new parameter from one GPU to others, but the GPU is an async device, it means that when the calling returned on CPU side, the tasks in the GPU are not finished, this will induce, in the next iteration, the parameters in some GPUs may be the old value but not the updated.